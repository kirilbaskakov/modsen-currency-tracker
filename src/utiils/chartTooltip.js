const tooltipTitle = items => {
  const { x } = items[0].raw;
  const date = new Date(x).toLocaleDateString();
  return date;
};

const tooltipBody = items => {
  const { o, c, h, l } = items[0].raw;
  return `Open: ${o}\nClosed: ${c}\nHigh: ${h}\nLow: ${l}`;
};

export default {
  intersect: false,
  callbacks: {
    title: tooltipTitle,
    beforeBody: tooltipBody,
    label: () => ''
  }
};
