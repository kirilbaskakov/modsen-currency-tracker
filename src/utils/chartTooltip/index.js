import chartTooltip from './chartTooltip';

export default chartTooltip;
