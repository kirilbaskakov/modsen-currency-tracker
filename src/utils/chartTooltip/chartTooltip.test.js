import chartTooltip from './chartTooltip';

const items = [
  {
    raw: {
      x: new Date('23-05-2017'),
      o: 5,
      c: 10,
      l: 3,
      h: 15
    }
  }
];
test('tooltip title match snapshot', () => {
  const title = chartTooltip.callbacks.title(items);
  expect(title).toMatchSnapshot();
});

test('tooltip body match snapshot', () => {
  const bodyText = chartTooltip.callbacks.beforeBody(items);
  expect(bodyText).toMatchSnapshot();
});
