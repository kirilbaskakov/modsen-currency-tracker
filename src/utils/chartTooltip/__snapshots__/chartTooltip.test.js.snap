// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tooltip body match snapshot 1`] = `
"Open: 5
Closed: 10
High: 15
Low: 3"
`;

exports[`tooltip title match snapshot 1`] = `"Invalid Date"`;
